From b50321e3b9496634be5e5610cdc926ebb67d9fbd Mon Sep 17 00:00:00 2001
From: q66 <q66@chimera-linux.org>
Date: Mon, 9 Sep 2024 17:53:26 +0200
Subject: [PATCH 05/14] get rid of wrong linkage

There is no good way to compute the clang builtins library
path from inside the rust code, but there should be no harm
in just disabling nodefaultlibs for our platform.
---
 vendor/libc-0.2.155/src/unix/mod.rs |  7 +------
 vendor/libc-0.2.158/src/unix/mod.rs |  7 +------
 vendor/libc-0.2.164/src/unix/mod.rs |  7 +------
 vendor/libc-0.2.168/src/unix/mod.rs | 15 +--------------
 vendor/libc-0.2.169/src/unix/mod.rs | 15 +--------------
 vendor/libc-0.2.171/src/unix/mod.rs | 15 +--------------
 vendor/libc-0.2.172/src/unix/mod.rs | 15 +--------------
 vendor/libc-0.2.174/src/unix/mod.rs | 15 +--------------
 8 files changed, 8 insertions(+), 88 deletions(-)

diff --git a/vendor/libc-0.2.155/src/unix/mod.rs b/vendor/libc-0.2.155/src/unix/mod.rs
index 49984d3f0..76890a787 100644
--- a/vendor/libc-0.2.155/src/unix/mod.rs
+++ b/vendor/libc-0.2.155/src/unix/mod.rs
@@ -348,12 +348,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", kind = "static", modifiers = "-bundle",
-                        cfg(target_feature = "crt-static")))]
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", cfg(not(target_feature = "crt-static"))))]
-        extern {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
diff --git a/vendor/libc-0.2.158/src/unix/mod.rs b/vendor/libc-0.2.158/src/unix/mod.rs
index 8b3d988ae..5ac630248 100644
--- a/vendor/libc-0.2.158/src/unix/mod.rs
+++ b/vendor/libc-0.2.158/src/unix/mod.rs
@@ -348,12 +348,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", kind = "static", modifiers = "-bundle",
-                        cfg(target_feature = "crt-static")))]
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", cfg(not(target_feature = "crt-static"))))]
-        extern {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
         extern {}
diff --git a/vendor/libc-0.2.164/src/unix/mod.rs b/vendor/libc-0.2.164/src/unix/mod.rs
index db60f8ef2..a86e566b5 100644
--- a/vendor/libc-0.2.164/src/unix/mod.rs
+++ b/vendor/libc-0.2.164/src/unix/mod.rs
@@ -385,12 +385,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", kind = "static", modifiers = "-bundle",
-                        cfg(target_feature = "crt-static")))]
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", cfg(not(target_feature = "crt-static"))))]
-        extern {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
diff --git a/vendor/libc-0.2.168/src/unix/mod.rs b/vendor/libc-0.2.168/src/unix/mod.rs
index a14dafdf0..d15249951 100644
--- a/vendor/libc-0.2.168/src/unix/mod.rs
+++ b/vendor/libc-0.2.168/src/unix/mod.rs
@@ -445,20 +445,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(
-                name = "c",
-                kind = "static",
-                modifiers = "-bundle",
-                cfg(target_feature = "crt-static")
-            )
-        )]
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(name = "c", cfg(not(target_feature = "crt-static")))
-        )]
-        extern "C" {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
diff --git a/vendor/libc-0.2.169/src/unix/mod.rs b/vendor/libc-0.2.169/src/unix/mod.rs
index a14dafdf0..d15249951 100644
--- a/vendor/libc-0.2.169/src/unix/mod.rs
+++ b/vendor/libc-0.2.169/src/unix/mod.rs
@@ -445,20 +445,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(
-                name = "c",
-                kind = "static",
-                modifiers = "-bundle",
-                cfg(target_feature = "crt-static")
-            )
-        )]
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(name = "c", cfg(not(target_feature = "crt-static")))
-        )]
-        extern "C" {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
diff --git a/vendor/libc-0.2.171/src/unix/mod.rs b/vendor/libc-0.2.171/src/unix/mod.rs
index b2de87ebf..488d63593 100644
--- a/vendor/libc-0.2.171/src/unix/mod.rs
+++ b/vendor/libc-0.2.171/src/unix/mod.rs
@@ -450,20 +450,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(
-                name = "c",
-                kind = "static",
-                modifiers = "-bundle",
-                cfg(target_feature = "crt-static")
-            )
-        )]
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(name = "c", cfg(not(target_feature = "crt-static")))
-        )]
-        extern "C" {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
diff --git a/vendor/libc-0.2.172/src/unix/mod.rs b/vendor/libc-0.2.172/src/unix/mod.rs
index 433eeec95..c281c0151 100644
--- a/vendor/libc-0.2.172/src/unix/mod.rs
+++ b/vendor/libc-0.2.172/src/unix/mod.rs
@@ -470,20 +470,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(
-                name = "c",
-                kind = "static",
-                modifiers = "-bundle",
-                cfg(target_feature = "crt-static")
-            )
-        )]
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(name = "c", cfg(not(target_feature = "crt-static")))
-        )]
-        extern "C" {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
diff --git a/vendor/libc-0.2.174/src/unix/mod.rs b/vendor/libc-0.2.174/src/unix/mod.rs
index 96209e7e7..777f5f30e 100644
--- a/vendor/libc-0.2.174/src/unix/mod.rs
+++ b/vendor/libc-0.2.174/src/unix/mod.rs
@@ -478,20 +478,7 @@ cfg_if! {
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern "C" {}
     } else if #[cfg(any(target_env = "musl", target_env = "ohos"))] {
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(
-                name = "c",
-                kind = "static",
-                modifiers = "-bundle",
-                cfg(target_feature = "crt-static")
-            )
-        )]
-        #[cfg_attr(
-            feature = "rustc-dep-of-std",
-            link(name = "c", cfg(not(target_feature = "crt-static")))
-        )]
-        extern "C" {}
+        // we don't set -nodefaultlibs, so no need to link anything
     } else if #[cfg(target_os = "emscripten")] {
         // Don't pass -lc to Emscripten, it breaks. See:
         // https://github.com/emscripten-core/emscripten/issues/22758
-- 
2.50.1

